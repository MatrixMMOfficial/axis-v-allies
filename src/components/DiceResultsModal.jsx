import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import Dice from './Dice'

class DiceResultsModal extends Component {
  constructor () {
    super()
    this.state = {
      reveal: false
    }
    this.rolledCount = 0
    this.reveal = this.reveal.bind(this)
  }

  reveal () {
    this.rolledCount += 1
    if (this.rolledCount === this.props.rolls.length) {
      setTimeout(this.setState.bind(this, { reveal: true }), 500)
    }
  }

  render () {
    return (
      <div>
        <h1>{this.props.title} Results</h1>
        <Dice
          rolls={this.props.rolls}
          handleReveal={this.reveal}
          goalFunction={this.props.goalFunction} />
        {this.state.reveal && this.props.children}
      </div>
    )
  }
}

export default DiceResultsModal

