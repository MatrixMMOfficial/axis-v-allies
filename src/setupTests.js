import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

export const mockRandom = fn => {
  const mockMath = Object.create(global.Math)
  mockMath.random = fn
  global.Math = mockMath
}
