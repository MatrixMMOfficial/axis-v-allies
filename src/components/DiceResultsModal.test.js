'use strict'

import React from 'react'
import DiceResultsModal from './DiceResultsModal'
import renderer from 'react-test-renderer'
import { mockRandom } from '../setupTests'

mockRandom(() => 0.5)

it('renders correctly', () => {
  const tree = renderer.create(
    <DiceResultsModal rolls={[1]} goalFunction={n => n} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

