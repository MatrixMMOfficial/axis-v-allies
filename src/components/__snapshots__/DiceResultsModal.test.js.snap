// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
     Results
  </h1>
  <div>
    <div
      className="dieWrapper"
    >
      <div
        className="die"
        style={
          Object {
            "transition": "transform 3.6s",
          }
        }
      >
        <figure
          className="front"
        >
          <svg
            viewBox="0 0 170 170"
          >
            <circle
              cx="85"
              cy="85"
              r="24"
            />
          </svg>
        </figure>
        <figure
          className="bottom"
        >
          <svg
            viewBox="0 0 170 170"
          >
            <circle
              cx="140"
              cy="30"
              r="24"
            />
            <circle
              cx="30"
              cy="140"
              r="24"
            />
          </svg>
        </figure>
        <figure
          className="left"
        >
          <svg
            viewBox="0 0 170 170"
          >
            <circle
              cx="85"
              cy="85"
              r="24"
            />
            <circle
              cx="140"
              cy="30"
              r="24"
            />
            <circle
              cx="30"
              cy="140"
              r="24"
            />
          </svg>
        </figure>
        <figure
          className="right"
        >
          <svg
            viewBox="0 0 170 170"
          >
            <circle
              cx="140"
              cy="30"
              r="24"
            />
            <circle
              cx="30"
              cy="140"
              r="24"
            />
            <circle
              cx="30"
              cy="30"
              r="24"
            />
            <circle
              cx="140"
              cy="140"
              r="24"
            />
          </svg>
        </figure>
        <figure
          className="top"
        >
          <svg
            viewBox="0 0 170 170"
          >
            <circle
              cx="85"
              cy="85"
              r="24"
            />
            <circle
              cx="140"
              cy="30"
              r="24"
            />
            <circle
              cx="30"
              cy="140"
              r="24"
            />
            <circle
              cx="30"
              cy="30"
              r="24"
            />
            <circle
              cx="140"
              cy="140"
              r="24"
            />
          </svg>
        </figure>
        <figure
          className="back"
        >
          <svg
            viewBox="0 0 170 170"
          >
            <circle
              cx="140"
              cy="30"
              r="24"
            />
            <circle
              cx="30"
              cy="140"
              r="24"
            />
            <circle
              cx="30"
              cy="30"
              r="24"
            />
            <circle
              cx="140"
              cy="140"
              r="24"
            />
            <circle
              cx="30"
              cy="85"
              r="24"
            />
            <circle
              cx="140"
              cy="85"
              r="24"
            />
          </svg>
        </figure>
      </div>
    </div>
    <svg
      height="50"
      version="1.1"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <radialGradient
          fx="30%"
          fy="30%"
          id="dieGradient"
        >
          <stop
            offset="5%"
            stopColor="#333"
          />
          <stop
            offset="95%"
            stopColor="#000"
          />
        </radialGradient>
      </defs>
    </svg>
  </div>
</div>
`;
